exasol_sphinx_github_pages_generator/deploy_github_pages.py

```diff
@@ -72,16 +67,23 @@ def checkout_target_branch_as_worktree(target_branch, worktree, push_origin):
             run(["git", "reset", "--hard", gh_pages_root_branch])
             print(f"Delete root branch {gh_pages_root_branch}")
             run(["git", "branch", "-D", gh_pages_root_branch])
-        # popd ?
+        os.chdir(currentworkdir)
 
-
-def build_and_copy_documentation(build_dir, worktree, source_branch, source_dir):
+def build_and_copy_documentation(build_dir, worktree, source_branch, source_dir, module_path):
     print("Build with sphinx")
-    run(["sphinx-build", "-M", "html", source_dir, build_dir, "-W"])
+    currentworkdir = os.getcwd()
+    print("currentworkdir :" + currentworkdir)
+    # automatically generates Sphinx sources inside the "api" directory that document
+    # the package found in "module_path"
+    # -T: not table of contents
+    # -e: put documentation for ech module on own page
+    run(["sphinx-apidoc", "-T", "-e", "-o", "api", module_path])
+    # Builds the Sphinx documentation. Generates html files inside "build_dir" using "source_dir"
+    # -W: Turns warnings into errors
+    run(["sphinx-build", "-b", "html", "-W", source_dir, build_dir])
```

Comment:
I believe you don't check the return-code of any of the run-methods. That will make error detection quiete hard if something goes wrong.
You could wrap the run-method in your own run-method which checks the returncode and raises an exception if the return code is not 0.